include/nihstro/shader_bytecode.h

```diff
@@ -379,47 +379,68 @@ struct OpCode {
-        case Id::MIN:     return { Type::Arithmetic,         Info::TwoArguments,        "min" };
-        case Id::RCP:     return { Type::Arithmetic,         Info::OneArgument,         "rcp" };
-        case Id::RSQ:     return { Type::Arithmetic,         Info::OneArgument,         "rsq" };
-        case Id::MOVA:    return { Type::Arithmetic,         Info::MOVA,                "mova" };
-        case Id::MOV:     return { Type::Arithmetic,         Info::OneArgument,         "mov" };
-        case Id::DPHI:    return { Type::Arithmetic,         Info::TwoArguments | Info::SrcInversed, "dphi" };
-        case Id::SGEI:    return { Type::Arithmetic,         Info::TwoArguments | Info::SrcInversed, "sgei" };
-        case Id::SLTI:    return { Type::Arithmetic,         Info::TwoArguments | Info::SrcInversed, "slti" };
-        case Id::NOP:     return { Type::Trivial,            0,                         "nop" };
-        case Id::END:     return { Type::Trivial,            0,                         "end" };
-        case Id::BREAKC:  return { Type::Conditional,        Info::BREAKC,              "breakc" };
-        case Id::CALL:    return { Type::Conditional,        Info::CALL,                "call" };
-        case Id::CALLC:   return { Type::Conditional,        Info::CALLC,               "callc" };
-        case Id::CALLU:   return { Type::UniformFlowControl, Info::CALLU,               "callu" };
-        case Id::IFU:     return { Type::UniformFlowControl, Info::IFU,                 "ifu" };
-        case Id::IFC:     return { Type::Conditional,        Info::IFC,                 "ifc" };
-        case Id::LOOP:    return { Type::UniformFlowControl, Info::LOOP,                "loop" };
-        case Id::EMIT:    return { Type::Trivial,            0,                         "emit" };
-        case Id::SETEMIT: return { Type::SetEmit,            0,                         "setemit" };
-        case Id::JMPC:    return { Type::Conditional,        Info::JMPC,                "jmpc" };
-        case Id::JMPU:    return { Type::Conditional,        Info::JMPU,                "jmpu" };
-        case Id::CMP:     return { Type::Arithmetic,         Info::Compare,             "cmp" };
-        case Id::MADI:    return { Type::MultiplyAdd,        Info::SrcInversed,         "madi" };
-        case Id::MAD:     return { Type::MultiplyAdd,        0,                         "mad" };
-
-        default:
-            return { Type::Unknown, 0, "UNK" };
-        }
+    const Info& GetInfo() const {
+        static OpCode::Info info_table[] =
+        {
+            { OpCode::Type::Arithmetic, OpCode::Info::TwoArguments, "add" }
+            , { OpCode::Type::Arithmetic, OpCode::Info::TwoArguments, "dp3" }
+            , { OpCode::Type::Arithmetic, OpCode::Info::TwoArguments, "dp4" }
+            , { OpCode::Type::Arithmetic, OpCode::Info::TwoArguments, "dph" }
+            , { OpCode::Type::Unknown, 0, "UNK" }
```

Comment:
Please use an `static OpCode::Info unknown_instruction` object instead of repeating this line all over the place.